File: algorithms/168.Excel_Sheet_Column_Title/Excel_Sheet_Column_Title.cpp
<<<<<<< HEAD
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    string convertToTitle(int n) {
        string ret = "";
        --n;
        while (n >= 0) {
            ret += (char)('A' + n % 26);
            n /= 26;
            --n;
        }
        reverse(ret.begin(), ret.end());
        return ret;
    }
};

class Solution {
 public:
  string convertToTitle(int n) {
    string result;
    while(n >= 1) {
      result.push_back((n - 1) % 26 + 'A');
      n = (n - 1) / 26;
    }
    reverse(result.begin(), result.end());
    return result;
  }
};
class Solution {
 public:
  string convertToTitle(int n) {
    string result;
    while(n >= 1) {
      result.push_back((n - 1) % 26 + 'A');
      n = (n - 1) / 26;
    }
    reverse(result.begin(), result.end());
    return result;
  }
};
int main() {
    int n;
    Solution s;
    while (cin >> n)
        cout << s.convertToTitle(n) << endl;
    return 0;
}

=======
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    string convertToTitle(int n) {
        string ret = "";
        --n;
        while (n >= 0) {
            ret += (char)('A' + n % 26);
            n /= 26;
            --n;
        }
        reverse(ret.begin(), ret.end());
        return ret;
    }
};

class Solution {
 public:
  string convertToTitle(int n) {
    string result;
    while(n >= 1) {
      result.push_back((n - 1) % 26 + 'A');
      n = (n - 1) / 26;
    }
    reverse(result.begin(), result.end());
    return result;
  }
};
class Solution {
 public:
  string convertToTitle(int n) {
    string result;
    while(n >= 1) {
      result.push_back((n - 1) % 26 + 'A');
      n = (n - 1) / 26;
    }
    reverse(result.begin(), result.end());
    return result;
  }
};
int main() {
    int n;
    Solution s;
    while (cin >> n)
        cout << s.convertToTitle(n) << endl;
    return 0;
}

>>>>>>> 6200c8704614e918c8bfa5357c648dd1b4f7eb74

File: algorithms/168.Excel_Sheet_Column_Title/Excel_Sheet_Column_Title.java


import java.util.Scanner;

public class Solution {

    public String convertToTitle(int n) {
        StringBuilder ret = new StringBuilder();
        while (n > 0) {
            ret.append((char)('A' + (n - 1) % 26));
            n = (n - 1) / 26;
        }
        return ret.reverse().toString();
    }

    // debug
    public static void main(String[] args) {
        Scanner cin = new Scanner(System.in);
        Solution s = new Solution();
        System.out.println(s.convertToTitle(26));
    }

}
/* Excel Sheet Column Title */
/* O£¨1£©space
 * O(log26(N)) time
 */
public class Solution {
    public String convertToTitle(int n) {
        if (n <= 0) return "";
        
        StringBuilder builder = new StringBuilder();
        
        while ( n > 0) {
        	if ( n <= 26) {
        		builder.insert(0, (char)('A' + (n-1)));
        		break;
        	} else {
        		int v = (n-1) % 26;
           		builder.insert(0, (char)('A' + v));
           	    n = (n-1)/26;
        	}
        	
        }
        
        return builder.toString();
    }
    
    
    public String convertToTitle2(int n) {
        if (n <= 0) return "";
        
        StringBuilder builder = new StringBuilder();
        
        while ( n > 0) {
       		int v = (n-1) % 26;
       		builder.insert(0, (char)('A' + v));
       	    n = (n-1)/26;
        }
        
        return builder.toString();
    }
    
    
}